File: 832-FlippingAnImage/C++/main.cpp
class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& A) {
        int r = A.size(), c = A[0].size();
        for (int y = 0; y < r; y++) {
            for (int x = 0; x < std::ceil( (float) c / 2.0); x++) {
                int temp1 = A[y][x], temp2 = A[y][c-1-x];
                A[y][x] = 1-temp2;
                A[y][c-1-x] = 1-temp1;
            }
        }
        
        return A;
            
    }
};

File: 832-FlippingAnImage/Python/main.py
class Solution:
    # inplace
    def flipAndInvertImage(self, A):
        """
        :type A: List[List[int]]
        :rtype: List[List[int]]
        """
        r, c = len(A), len(A[0])
        for y in range(r):
            for x in range(math.ceil(c/2)):
                A[y][x], A[y][c-1-x] = 1-A[y][c-1-x], 1-A[y][x]
        
        return A
        
    
    def flipAndInvertImage2(self, A):
        """
        :type A: List[List[int]]
        :rtype: List[List[int]]
        """
        return [[1-e for e in row[::-1]] for row in A]